### PR TITLE
Introduce theme context

### PR DESCRIPTION
**Pull request by Nefsen402**
And remove shared

A demo is in examples/index.jsx

Based on: #4

### CODE REVIEW

File: src/components/Markdown.jsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ const renderList = (lines) => {
     return nodes;
 };
 
-const Element = ({ each: e }) => {
+const Element = Theme.use(theme => ({ each: e }) => {
```
Comment (torrinworx) [inline src/components/Markdown.jsx:91]: The `each` prop seems to not be working here, seems to only be returning the last item in an OArray the given number of indexes in the array

Comment (Nefsen402) [reply src/components/Markdown.jsx:91]: Ah yes, that's fixed.

**Comment by torrinworx** [comment]
Nice, LGTM, are there still things your working on or is it good to merge?

**Comment by Nefsen402** [comment]
This PR would be a draft if I was still working on it. Nothing more for now.

**Pull request merged**